common/djangoapps/student/views.py

```diff
@@ -1793,21 +1795,33 @@ def create_account_with_params(request, params):
             'email_from_address',
             settings.DEFAULT_FROM_EMAIL
         )
-        try:
-            if settings.FEATURES.get('REROUTE_ACTIVATION_EMAIL'):
-                dest_addr = settings.FEATURES['REROUTE_ACTIVATION_EMAIL']
-                message = ("Activation for %s (%s): %s\n" % (user, user.email, profile.name) +
-                           '-' * 80 + '\n\n' + message)
-                mail.send_mail(subject, message, from_address, [dest_addr], fail_silently=False)
-            else:
-                user.email_user(subject, message, from_address)
-        except Exception:  # pylint: disable=broad-except
-            log.error(
-                u'Unable to send activation email to user from "%s" to "%s"',
-                from_address,
-                dest_addr,
-                exc_info=True
-            )
+        while True:
+            try:
+                if settings.FEATURES.get('REROUTE_ACTIVATION_EMAIL'):
+                    dest_addr = settings.FEATURES['REROUTE_ACTIVATION_EMAIL']
+                    message = ("Activation for %s (%s): %s\n" % (user, user.email, profile.name) +
+                               '-' * 80 + '\n\n' + message)
+                    mail.send_mail(subject, message, from_address, [dest_addr], fail_silently=False)
+                else:
+                    user.email_user(subject, message, from_address)
+                # Log that the Activation Email has been sent to user without an exception
+                log.info("Activataion Email has been sent to User {user_email}".format(
+                    user_email=dest_addr
+                ))
+                break
+            except Exception:  # pylint: disable=broad-except
+                retries += 1
+                if retries > settings.FEATURES.get('RETRY_MAX_ATTEMPTS'):
+                    break
+                log.error(
+                    u'Unable to send activation email to user from "%s" to "%s"',
+                    from_address,
+                    dest_addr,
+                    exc_info=True
+                )
+            if retries > 1:
+                time.sleep(settings.RETRY_TIMEOUT)
```

Comment:
this is always dangerous to do in process (and not in a background task). What's the retry timeout right now?

Would this make more sense as a background task? Like abstract the try/except block into a function `send_user_email` or something and turn it into a celery task?